### PR TITLE
WIP: Optionally truncate long data in SDF output

### LINKED ISSUE #630: SDF writer does not adhere to 200 character limit for data field values
I'm interested to know what you think is the correct behaviour if I ask the SD writer to write out a data field value that has a line containing more than 200 characters.. The docs say:

"A [Data] value can extend over multiple lines containing up to 200 characters each."

Is the max 199 or 200? I don't know. My question is really about how this should be handled by a writer. Truncated? Truncated and split over several lines (to be joined on reading?)? Handled as-is but with a warning? I do think the current behaviour needs changing (handled as-is but with no warning...I think) as it can lead to trouble with downstream tools.

### PR DESCRIPTION
**Pull request by johnmay**
A possibility to resolve #630

### CODE REVIEW

**Comment by johnmay** [comment]
Think this is okay to merge if you aggressively Egon. Could also flip default to always truncate but I think the vast majority of other tools won’t care about >200 chars

**Comment by egonw** [comment]
Not sure how to interpret "if you aggressively Egon". I'm happy to wait...

**Comment by johnmay** [comment]
Sorry meant to say **agree** - auto correct coupled with huge workload at the moment.

**Comment by johnmay** [comment]
I think I typed agres and it out correct... haha

File: storage/ctab/src/test/java/org/openscience/cdk/io/SDFWriterTest.java [hunk 1]
```diff
@@ -370,4 +370,33 @@ public void setProgramName() {
...
+                            "ThisIsAVeryLongFieldThatShouldBeWrapped" +
+                            "ThisIsAVeryLongFieldThatShouldBeWrapped" +
+                            "ThisIsAVeryLongFieldThatShouldBeWrapped");
+            sdfw.write(mol);
+        } catch (IOException | CDKException e) {
+            e.printStackTrace();
+        }
+        String sdf = sw.toString();
+        assertThat(sdf,
+                   CoreMatchers.containsString("ThisIsAVeryLongFieldThatShouldBeWrappedThisIsAVeryLongFieldThatShouldBeWrappedThisIsAVeryLongFieldThatShouldBeWrappedThisIsAVeryLongFieldThatShouldBeWrappedThisIsAVeryLongFieldThatShouldBeWrappedThisI\n"));
```
Comment (egonw) [inline storage/ctab/src/test/java/org/openscience/cdk/io/SDFWriterTest.java:400]: If truncation is set to true, would we not expect newlines in the string?

File: storage/ctab/src/test/java/org/openscience/cdk/io/SDFWriterTest.java [hunk 2]
```diff
@@ -370,4 +370,33 @@ public void setProgramName() {
...
         }
     }
+
+    @Test
+    public void optionallyTruncateLongProperties() {
+        StringWriter sw = new StringWriter();
+        try (SDFWriter sdfw = new SDFWriter(sw)) {
+            sdfw.getSetting(MDLV2000Writer.OptWriteDefaultProperties)
+                .setSetting("false");
+            sdfw.getSetting(SDFWriter.OptTruncateLongData)
```
Comment (egonw) [inline storage/ctab/src/test/java/org/openscience/cdk/io/SDFWriterTest.java:380]: Can you also add the mirror test, with the expected outcome when it is set to false? Also relates to my below question.

File: storage/ctab/src/test/java/org/openscience/cdk/io/SDFWriterTest.java (hunk 1 above)
Comment (johnmay) [reply storage/ctab/src/test/java/org/openscience/cdk/io/SDFWriterTest.java:400]: truncate not split, there is nothing in the spec about how to join long fields back together - only for RDfile.

Comment (egonw) [reply storage/ctab/src/test/java/org/openscience/cdk/io/SDFWriterTest.java:400]: Thanks for the clarification!

**Pull request merged**